agent/structs/structs_oss.go

```diff
@@ -126,6 +127,10 @@ func (n ServiceName) String() string {
 	return n.Name
 }
 
+func (n ServiceName) DiscoveryChainID() string {
+	return fmt.Sprintf("%s?dc=dc2", n.String())
+}
```

Comment:
This appears to only be used by tests, so would probably be better as a function in `state_test.go` instead of a method on `ServiceName`.

Reply:
Thanks @dnephin! I've moved it to the test file and left the hardcoding of the dc and named the function appropriately since it's the only case the function is used for. Let me know if you'd prefer a different style.